extensions/AMD/SPV_AMDX_shader_enqueue.asciidoc

```diff
@@ -268,39 +380,85 @@ _Node Index_
 
 [cols="3,1,1"]
 |======
-2+|[[OpFinalizeNodePayloadsAMDX]]*OpFinalizeNodePayloadsAMDX* +
+2+|[[OpEnqueueNodePayloadsAMDX]]*OpEnqueueNodePayloadsAMDX* +
  +
-Optionally indicates that all accesses to an array of output payloads have completed.
+Enqueues a previously allocated payload array for execution by its node. +
  +
-_Payload Array_ is a payload array previously initialized by *OpInitializeNodePayloadsAMDX*.
+_Payload Array_ is a pointer to a payload array that was previously allocated by *OpAllocateNodePayloadsAMDX*. +
  +
-This instruction must be called in uniform control flow.
- +
-_Payload Array_ must be an *OpTypePointer* with a _Storage Class_ of _OutputNodePayloadAMDX_, and a _Type_ of *OpTypeArray* or *OpTypeRuntimeArray* with an _Element Type_ of *OpTypeStruct*.
-_Payload Array_ must not have been previously finalized by *OpFinalizeNodePayloadsAMDX*.
+This instruction must be called in uniform control flow within the workgroup.
 1+|Capability: +
 *ShaderEnqueueAMDX*
 | 2 | 5075
 | _<id>_ +
 _Payload Array_
 |======
 
+[cols="2*1,3*2",width="100%"]
+|=====
+4+|[[OpNodePayloadArrayLengthAMDX]]*OpNodePayloadArrayLengthAMDX* +
+ +
+Query the length of a payload array. Must only be used with input payload arrays or allocated output payload arrays. +
+ +
+_Result_ will be equal to the _Payload Count_ value used to allocate _Payload Array_, or to the number of received payloads if the shader is using *CoalescingAMDX* execution mode. Otherwise, _Result_ will be 1. +
+ +
+_Result Type_ must be an *OpTypeInt* with 32-bit _Width_ and 0 _Signedness_. +
+ +
+_Payload Array_ is a pointer to a payload array previously allocated by *OpAllocateNodePayloadsAMDX*, or declared via *OpVariable* in the *NodePayloadAMDX* storage class as an input.
+1+|<<Capability,Capability>>: +
+*Shader*
```

Comment:
https://github.com/KhronosGroup/SPIRV-Headers/pull/452 ties it to `ShaderEnqueueAMDX` not `Shader`